practice_2/NewsApp/View/NewsCollectionViewController.swift

```diff
@@ -0,0 +1,133 @@
+import UIKit
+
+class NewsCollectionViewController: UICollectionViewController {
+    private let itemsPerRow: CGFloat = 2
+    private let itemsPerColumn: CGFloat = 3
+    private let sectionInsets = UIEdgeInsets(top: 10, left: 10, bottom: 10, right: 10)
+    var presenter: NewsPresenterProtocol?
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setup()
+    }
+
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "detailsNewsSegue" {
+            guard let detailsNewsVC = segue.destination as? DetailsNewsViewController else { return }
+            guard let cell = sender as? NewsCell else { return }
+            detailsNewsVC.article = cell.article
+        }
+    }
+
+    private func setup() {
+        let networkService = NewsNetworkService.sharedNews
+        self.presenter = NewsPresenter(view: self, networkService: networkService)
+    }
+
+    @IBAction func slideMenuTapped(_ sender: UIBarButtonItem) {
+        HamburgerMenu().triggerSideMenu()
+    }
+
+    // MARK: Sort alphabetically
+    @IBAction func sortAction(_ sender: UIBarButtonItem) {
+        guard let title = sender.title else { return }
+        if title == "up" {
+            self.presenter?.sortAlphabetically(inOrder: title)
+            sender.title = "down"
+            sender.image = UIImage(systemName: "arrow.down")
```

Comment:
try move these system names to constants

Reply:
Thank you. Done